_episodes/04-changes.md

```diff
@@ -590,10 +590,11 @@ Date:   Thu Aug 22 09:51:46 2013 -0400
 >
 >    Note, our newly created empty directory `directory` does not appear in
 >    the list of untracked files even if we explicitly add it (_via_ `git add`) to our
->    repository. This is the reason why you will sometimes see `.gitkeep` files
->    in otherwise empty directories. Unlike `.gitignore`, these files are not special
->    and their sole purpose is to populate a directory so that Git adds it to
->    the repository. In fact, you can name such files anything you like.
+>    repository. This is due to Git only operating on files, and is sometimes worked-
+>    around by creating empty `.gitkeep` files if an otherwise empty folder is deemed 
+>    to be tracked by Git. However, it's better to avoid this and look for ways to 
+>    automatically create such folders when your code needs them, or try to very 
+>    briefly explain the need for the empty folder with a `.keep-because-...` file.
```

Comment:
I understand your intention here by explaining the reason for `.keep-because-` to make the reason for having the .gitkeep file obvious. But in the wild I haven't seen any projects do this. Also `.gitkeep` is complimentary in naming to `.gitignore`. I disagree with incorporating the change as it is worded here. The original text says you can name them whatever you'd like without telling learners what they should be doing.

Reply:
@katrinleinweber, will you clarify
> However, it's better to avoid this and look for ways to automatically create such
>  folders when your code needs them.

?

Reply:
@kekoziar & @scottgruber: How about incorporating [my clarification above](https://github.com/swcarpentry/git-novice/pull/571#issuecomment-452216390) in some fashion?

> For example, functions that create files but not the directory should be identified and improved, rather than pre-populating repositories with empty folders.